src/network/core_utils.rs

```diff
@@ -0,0 +1,468 @@
+                        return Err(std::io::Error::new(
+                            std::io::ErrorKind::Other,
+                            format!("failed to get veth interface {}: {}", &container_veth, err),
+                        ))
+                    }
+                }
+            }
+            Err(err) => {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    format!("failed to open namespace: {}", err),
+                ))
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn configure_netns_interface(
+        ifname: &str,
+        ip_add: &str,
+        ip_mask: &str,
+        gw_ip_add: &str,
+    ) -> Result<(), Error> {
+        return executor::block_on(CoreUtils::configure_netns_interface_async(
+            ifname, ip_add, ip_mask, gw_ip_add,
+        ));
+    }
+
+    async fn configure_netns_interface_async(
+        ifname: &str,
+        ip_add: &str,
+        ip_mask: &str,
+        gw_ip_add: &str,
+    ) -> Result<(), Error> {
+        let (_connection, handle, _) = match rtnetlink::new_connection() {
+            Ok((conn, handle, messages)) => (conn, handle, messages),
+            Err(err) => {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    format!("failed to connect: {}", err),
+                ))
+            }
+        };
+
+        // ip netns exec ip link set <ifname> up
+        if let Err(err) = CoreUtils::set_link_up(&handle, ifname).await {
+            return Err(err);
+        }
+
+        // ip netns exec <namespace> ip addr add <addr>/<mask> dev <ifname>
+        match format!("{}/{}", ip_add, ip_mask).parse() {
+            Ok(ip) => {
+                if let Err(err) = CoreUtils::add_ip_address(&handle, ifname, &ip).await {
+                    return Err(err);
+                }
+            }
+            Err(err) => {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    format!(
+                        "failed to parse address {}: {}",
+                        format!("{}/{}", ip_add, ip_mask),
+                        err
+                    ),
+                ))
+            }
+        }
+
+        // ip netns exec <namespace> ip route add default via <gateway> dev <ifname>
+        match gw_ip_add.to_string().parse() {
+            Ok(gateway) => match "0.0.0.0/0".to_string().parse() {
+                Ok(dest) => {
+                    if let Err(err) = CoreUtils::add_route(&handle, &dest, &gateway).await {
+                        return Err(err);
+                    }
+                }
+                Err(err) => {
+                    return Err(std::io::Error::new(
+                        std::io::ErrorKind::Other,
+                        format!("failed to parse address 0.0.0.0/0: {}", err),
+                    ))
+                }
+            },
```

Comment:
You have to check if you have an ipv4 or ipv6 address and add the correct route.